src/OutSystems/Maps/MapAPI/MarkerManager.ts

```diff
@@ -140,20 +168,24 @@ namespace OutSystems.Maps.MapAPI.MarkerManager {
 	 */
 	export function CreateMarker(mapId: string, markerId: string, configs: string): OSFramework.Maps.Marker.IMarker {
 		const map = MapManager.GetMapById(mapId, true);
-		if (!map.hasMarker(markerId)) {
-			const _marker = Provider.Maps.Google.Marker.MarkerFactory.MakeMarker(
-				map,
-				markerId,
-				OSFramework.Maps.Enum.MarkerType.Marker,
-				JSON.parse(configs)
-			);
-			markerArr.push(_marker);
-			markerMap.set(markerId, map.uniqueId);
-			map.addMarker(_marker);
-
-			return _marker;
+		if (map.providerType === OSFramework.Maps.Enum.ProviderType.Google) {
```

Comment:
This means that creating markers is not supported for Leaflet anymore or am I understanding it wrong?

Reply:
Correct. The Method is **only** used in the private client action `CreateMarker` that is used in the private client action `ChangeMarkers` that is used in the `StaticMap` block, that is only supported by google maps. 
The only thing I did was making the validation explicit so that at least an error is thrown in case the JS API is misused.